build/common.project.props

```diff
@@ -124,7 +124,9 @@
     <MinimumVisualStudioVersion>15.0</MinimumVisualStudioVersion>
     <VsixOutputDirName>VS15</VsixOutputDirName>
     <CodeAnalysisRuleSet Condition="'$(CodeAnalysisRuleSet)' == ''">$(BuildCommonDirectory)NuGet.ruleset</CodeAnalysisRuleSet>
-    <EnforceCodeStyleInBuild Condition=" '$(EnforceCodeStyleInBuild)' != '' " >true</EnforceCodeStyleInBuild>
+    <EnforceCodeStyleInBuild Condition="'$(EnforceCodeStyleInBuild)' == ''">true</EnforceCodeStyleInBuild>
+    <WarningsNotAsErrors>$(WarningsNotAsErrors);EnableGenerateDocumentationFile</WarningsNotAsErrors>
+    <AnalysisLevel>8.0</AnalysisLevel>
```

Comment:
I think the spirit of that is to only add it if you need it. 

See details about https://github.com/dotnet/designs/blob/main/proposed/sdk-analysis-level.md, that we collaborated on.

This is a team decision ofc, but I think in general, we want to get a steady stream of warnings, instead of all at once.

Reply:
You know, I added this and now we're getting more errors 😅

Reply:
Okay so if I run it with level "Latest" it gives a new set of errors.

But the docs say, "If your project targets .NET 5 or later, or if you've added the [AnalysisMode](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#analysismode) property, the default value is latest." 

It looks like we target net472 for most of our projects, so I guess that adds up. I'm going to remove it for now bc I don't want to go about fixing all those changes in all those projects 😅